src/lib/containers/entity/metadata/EntityModal.tsx

```diff
@@ -70,24 +70,25 @@ export const EntityModal: React.FC<EntityModalProps> = ({
   })
 
   let primaryAction
-  let tertiaryActions
+  let secondaryActions
 
   if (!entityBundle) {
     primaryAction = { skeleton: true }
-    tertiaryActions = undefined
+    secondaryActions = undefined
   } else {
-    // TODO: Determine if we're on the entity page to conditonally show this button
-    primaryAction = {
-      skeleton: false,
-      copy: `Open ${entityTypeToFriendlyName(entityBundle.entityType!)}`,
-      onClick: () =>
-        window.open(
-          `${getEndpoint(
-            BackendDestinationEnum.PORTAL_ENDPOINT,
-          )}#!Synapse:${entityId}`,
-          '_blank',
-          'noopener',
-        ),
+    if (!window.location.href.includes(entityId)) {
```

Comment:
Render a button that links to the entity page in Synapse. I think for now, checking if the current URL including the synID is probably good enough.